vm/tests/executor.rs

```diff
@@ -655,32 +654,13 @@ fn dummy_input() -> AssetTransferInput {
     }
 }
 
-#[test]
-fn timelock_invalid_type() {
```

Comment:
Why did you remove this test?

Reply:
Let's see this: `&[Instruction::Push(0), Instruction::ChkTimelock(5)]`
The ChkTimelock function has a argument of the TimelockType, so constant value(like u8 or int) is invalid.
In addition, TimelockType can only be expressed from 1 to 4(Block, BlockAge, Time, and TimeAge). The test code was removed because other types and values were not accepted as an argument.

Reply:
In byte code, however, the constant can be the argument. This code is to test that case.

vm/src/decoder.rs
```rust
    #[test]
    #[allow(non_snake_case)]
    fn ChkTimelock() {
        assert_eq!(decode(&[opcode::CHKTIMELOCK]), Err(DecoderError::ScriptTooShort));
        assert_eq!(decode(&[opcode::CHKTIMELOCK, 0]), Err(DecoderError::InvalidImmediateValue(0)));
        assert_eq!(decode(&[opcode::CHKTIMELOCK, 5]), Err(DecoderError::InvalidImmediateValue(5)));
        assert_eq!(decode(&[opcode::CHKTIMELOCK, 1]), Ok(vec![Instruction::ChkTimelock(TimelockType::Block)]));
        assert_eq!(decode(&[opcode::CHKTIMELOCK, 2]), Ok(vec![Instruction::ChkTimelock(TimelockType::BlockAge)]));
        assert_eq!(decode(&[opcode::CHKTIMELOCK, 3]), Ok(vec![Instruction::ChkTimelock(TimelockType::Time)]));
        assert_eq!(decode(&[opcode::CHKTIMELOCK, 4]), Ok(vec![Instruction::ChkTimelock(TimelockType::TimeAge)]));

        assert_eq!(decode(&[opcode::CHKTIMELOCK, 1, opcode::CHKTIMELOCK]), Err(DecoderError::ScriptTooShort));
        assert_eq!(
            decode(&[opcode::CHKTIMELOCK, 1, opcode::CHKTIMELOCK, 2]),
            Ok(vec![Instruction::ChkTimelock(TimelockType::Block), Instruction::ChkTimelock(TimelockType::BlockAge)])
        );
    }
```

Reply:
Okay. I got it.